src/html-conventions-lib/general-connectors-html-convention.xsl

```diff
@@ -688,4 +688,133 @@
         </xsl:if>
     </xsl:template>
 
+    <xd:doc>
+        <xd:desc>[common-name-unique] - The connector role name $value$ is not unique. </xd:desc>
+        <xd:param name="connector"/>
+    </xd:doc>
+    <xsl:template name="connectorUniqueName">
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($sourceRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($sourceRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
+                        f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is not unique.'))
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+        <xsl:if test="boolean($targetRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($targetRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($targetRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
+                        f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is not unique.'))
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>[association-dependency-checker] - Connector role names are used in other
+            connectors </xd:desc>
+        <xd:param name="connector"/>
+        <xd:param name="isDependency"/>
+    </xd:doc>
+    <xsl:template name="connectorRoleCheck">
+        <xsl:param name="isDependency"/>
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:choose>
+                <xsl:when test="$isDependency = fn:true()">
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Association']"/>
```

Comment:
why when "is dependecy == true" you select "Association"

This logic looks counter unintuitive. Please refactor. the isDependecy flag shows that more tahn one functionality is merged into the function. Break it down.

Reply:
if the connector given as parameter is a Dependency you will want to look at Associations if they use the role name from the given Dependency. If this will be a specific checker I will break it down. TBD. Let me know the decision